p2panda-net/src/engine/topic_discovery.rs

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use anyhow::Result;
+use iroh_net::NodeId;
+use rand::random;
+use serde::{Deserialize, Serialize};
+use tokio::sync::mpsc;
+
+use crate::bytes::{FromBytes, ToBytes};
+use crate::engine::address_book::AddressBook;
+use crate::engine::constants::JOIN_PEERS_SAMPLE_LEN;
+use crate::engine::gossip::ToGossipActor;
+use crate::NetworkId;
+
+#[derive(Default, PartialEq, Eq)]
+enum Status {
+    #[default]
+    Idle,
+    Pending,
+    Active,
+}
+
+/// Manages the "topic discovery" background process.
+///
+/// Within a network peers can be interested in different topics. How topics are defined is up to
+/// the applications. Within a network there can be even multiple applications co-existing.
```

Comment:
These changes will still need a `cargo fmt` run (I'm making the suggestion in the GH UI to save some time).

```suggestion
/// Peers can be interested in different topics within a single network. How topics are defined is up to
/// the applications. Multiple applications can even co-exist within the same network.
```